R/utils.R

```diff
@@ -1,4 +1,8 @@
-where_at <- function(x, at, error_arg = caller_arg(at), error_call = caller_env()) {
+where_at <- function(x,
+                     at,
+                     user_env,
```

Comment:
It is a bit of a tough call, but when I add `user_env` to a helper that I know I'm going to be testing directly, I do `user_env = caller_env()` so that the user environment is "right" when `where_at()` is called directly at the top level (i.e. in a test). That way you don't have to touch the tests. This would also apply to `vctrs_vec_compat()`, which you also tweaked tests for, and then you may as well do it for `vctrs_list_compat()` too (if you decide to do it at all)

Reply:
This would _not_ fix the problem of having to set `user_env = current_env()` at the call sites of `where_at()` in the source code itself, but that seemed ok by me vs having to make the tests obscure looking

Reply:
Maybe using `user_env = caller_env()` is the answer to your question about how to test soft-deprecation 😅 , because that makes it work without any further tweaks